xds/internal/resolver/watch_service.go

```diff
@@ -160,6 +164,7 @@ func (w *serviceUpdateWatcher) updateVirtualHostsFromRDS(update xdsresource.Rout
 	}
 
 	w.lastUpdate.virtualHost = matchVh
+	w.lastUpdate.clusterSpecifierPlugins = update.ClusterSpecifierPlugins
```

Comment:
If this is the right place to put this, then I think the function needs a new name.

Reply:
I think this is the right place to put it (mechanical conversion from xdsclient struct -> xds resolver representation of data). Do you see any place else? The vh encapsulates the specific WRR cluster/csp name per route, and this is just the top level. The best name I could think of (after a few minutes of thinking haha) is updateStateFromRDS. Let me know if there's a better naming choice.

Reply:
I agree this is where it belongs.  It seems like this is the _real_ "handle RDS" function, and `handleRDSResp` should not do anything besides call this (and handle errors first), which is what it does.  Maybe rename to `applyRouteConfigUpdate`?

Reply:
Renamed to applyRouteConfigUpdate.